pkg/envoy/proxy.go

```diff
@@ -275,3 +282,14 @@ func (p *Proxy) buildCommand(ctx context.Context, cfgPath string) *exec.Cmd {
 
 	return cmd
 }
+
+// RemoveArgAndGetValue Function to get new args after removing given key
+// and also returns the value of key
+func RemoveArgAndGetValue(stringAr []string, key string) ([]string, string) {
+	for index, value := range stringAr {
+		if value == key {
+			return append(stringAr[:index], stringAr[index+2:]...), stringAr[index+1]
```

Comment:
Wouldn't `index+2` lead to index out of range panics here?

Reply:
no. https://go.dev/play/p/BhUVn3iaa1K

Reply:
Sorry, I also meant `index+1` in this case. Isn't this a possibility? https://go.dev/play/p/XZXT-RAHzTz

Reply:
log level should always have a value if key is provided.